cpp/validate_harvested_records.cc

```diff
@@ -322,58 +251,70 @@ void SendEmail(const std::string &email_address, const std::string &message_subj
 }
 
 
-void UpdateDB(DbConnection * const db_connection, const std::string &zeder_id, const std::string &zeder_instance,
-              const std::string &field_name, const std::string &field_presence_str)
-{
-    FieldPresence field_presence;
-    if (not StringToFieldPresence(field_presence_str, &field_presence))
-        LOG_ERROR("\"" + field_presence_str + "\" is not a valid field_presence!");
-    if (field_name.length() != MARC::Record::TAG_LENGTH)
-        LOG_ERROR("\"" + field_name + "\" is not a valid field name!");
-
-    db_connection->queryOrDie("UPDATE metadata_presence_tracer SET field_presence='" + field_presence_str + "' WHERE zeder_journal_id="
-                              + "(SELECT id FROM zeder_journals WHERE zeder_id=" + db_connection->escapeAndQuoteString(zeder_id) + " "
-                              + "AND zeder_instance=" + db_connection->escapeAndQuoteString(zeder_instance) + ") "
-                              + "AND field_name='" + field_name + "'");
-    if (db_connection->getNoOfAffectedRows() == 0)
-        LOG_ERROR("can't update non-existent database entry: " + zeder_id + "(" + zeder_instance + ")"
-                  + ", field_name: \"" + field_name + "\"");
+void CheckGenericRequirements(const MARC::Record &record, std::string * const reasons_for_being_invalid) {
+    if (not record.hasTag("001"))
```

Comment:
... might be more elegant to define a std::set<std::string> of the mandatory tags and iterate over it rather than copy/pasting code

Reply:
Done